src/jvm/test/org/jetbrains/kotlinx/lincheck_test/integration/AbstractIntegrationTest.kt

```diff
@@ -0,0 +1,101 @@
+
+        private fun isWindows(): Boolean = OS.contains("win")
+    }
+
+    abstract val projectPath: String
+
+    private fun buildGradleInitScriptToDumpTrace(
+        testClassName: String,
+        testMethodName: String,
+        fileToDump: File
+    ): String {
+        val pathToFatJar = File("build/libs/fat-trace-debugger-SNAPSHOT.jar")
+        return """
+            gradle.taskGraph.whenReady { TaskExecutionGraph taskGraph ->
+                def jvmTasks = taskGraph.allTasks.findAll { task -> task instanceof JavaForkOptions }
+                jvmTasks.each { task ->
+                    task.doFirst {
+                        def options = task as JavaForkOptions
+                        def jvmArgs = new ArrayList<>()
+                        jvmArgs.addAll(options.jvmArgs)
+                        jvmArgs.addAll([
+                            "-Dlincheck.traceDebuggerMode=true", 
+                            "-javaagent:${pathToFatJar.absolutePath}=$testClassName,$testMethodName,${fileToDump.absolutePath}"
+                        ])
+                        options.jvmArgs = jvmArgs
+                    }
+                }
+            }
+        """.trimIndent()
+    }
+
+    private fun getGolderDataPathFor(testClassName: String, testMethodName: String): String {
+        val projectName = File(projectPath).name
+        return File("").absolutePath + "/src/jvm/integrationTestData/$projectName/${testClassName}_$testMethodName.txt"
+    }
+
+    private fun createInitScriptAsTempFile(content: String): File {
+        val tempFile = File.createTempFile("initScript", ".gradle")
+        tempFile.writeText(content)
+        return tempFile
+    }
+
+    protected fun runGradleTest(
+        testClassName: String,
+        testMethodName: String,
+        vararg gradleCommand: String,
+    ) {
+        val tmpFile = File.createTempFile(testClassName + "_" + testMethodName, "")
+        val processBuilder = ProcessBuilder(
+            if (isWindows()) "gradlew.bat" else "./gradlew",
+            *gradleCommand,
+            "--tests",
+            "$testClassName.$testMethodName",
+            "--init-script",
+            createInitScriptAsTempFile(buildGradleInitScriptToDumpTrace(testClassName, testMethodName, tmpFile)).absolutePath,
+        )
+        processBuilder.directory(File(projectPath))
+
+        val process = processBuilder.start()
+        process.errorStream.copyTo(System.err)
+        process.waitFor().let {
+            if (it != 0) error("Gradle tests failed with exit code $it")
+        }
+
+        // TODO decide how to test: with gold data or run twice?
+        val goldDataFile = File(getGolderDataPathFor(testClassName, testMethodName))
+        if (goldDataFile.exists()) {
+            Assert.assertEquals(goldDataFile.readText(), tmpFile.readText())
+        } else {
+            copy(tmpFile, goldDataFile)
+            Assert.fail("The gold data file was created. Please rerun the test.")
```

Comment:
Let's fail test in this case. To overwrite the expected test output, let's use a gradle property. 
We already have one used for similar purposes in Lincheck representation tests, so we can re-use it here (it is called `overwriteRepresentationTestsOutput`). 
See the links below:

- https://github.com/JetBrains/lincheck/blob/a758c54e13058899488cb369207d6d24237cb19c/src/jvm/test/org/jetbrains/kotlinx/lincheck_test/util/TestUtils.kt#L54
- https://github.com/JetBrains/lincheck/blob/a758c54e13058899488cb369207d6d24237cb19c/src/jvm/test/org/jetbrains/kotlinx/lincheck_test/util/TestUtils.kt#L118
- https://github.com/JetBrains/lincheck/blob/a758c54e13058899488cb369207d6d24237cb19c/gradle.properties#L22

Reply:
What do you mean? The test will be failed with `Assert.fail`. This branch means that golden test data wasn't found and it will be created automatically. It is not exactly `overwrite`

Reply:
Oh, sorry, I missed the `Assert.fail`. 

Yet still, not sure that the golden data should be automatically created in this case. 
IMO, better have one single mechanism to overwrite golden data (e.g., when special flag is set), and in case when there is no file with golden data just output that file was not found.